### ISSUE TITLE
No vb.net code is generated, only c# code

### ISSUE THREAD

**Issue by MaciejJar** [dotnet/docfx#8590]
I have a project that has vb.net code but when generating the documentation, the code in it is changed to c# and I want it to continue to be vb.net. How do I set this up?

**Comment by yufeih** [comment]
This is currently not supported yet.

**Comment by JamesFiitWork** [comment]
Is there a guideline that helps us to add this feature to Docfx? I've found some metadata related to VB symbol name / declaration. But I couldn't determine the most natural way of implementing this feature.

As a .NET class library developer, I must document the APIs in both VB.NET and C# before selling the library to Japanese customers.

**Comment by Nukepayload2** [comment]
In your project template, you can replace `{{syntax.content.0.lang}}` with `{{syntax.content.1.lang}}` to enable VB declarations.

In some cases, VB declarations are wrong. I'll fork this repository to fix wrong VB declarations.

**Comment by Nukepayload2** [comment]
Affected by: https://github.com/dotnet/roslyn/issues/14683

`{{syntax.content.1.lang}}` produces incorrect result when:

- A parameter type is `ByRef`: `ByRef` will be removed.
- A property is default: The property name is incorrect.
- A method is `hidebysig`: `Overloads` will be removed.

**Comment by Nukepayload2** [comment]
My fork fixes VB declarations for frequently used syntaxes. If it's useful and acceptable for docfx v2, I can create a pull request.

https://github.com/Nukepayload2/docfx/tree/feature/vb-declaration-in-v2

**Comment by jamaa** [comment]
> In your project template, you can replace `{{syntax.content.0.lang}}` with `{{syntax.content.1.lang}}` to enable VB declarations.

@Nukepayload2 can you please explain where exactly this needs to be set? I couldn't find this option in the [config reference](https://dotnet.github.io/docfx/reference/docfx-json-reference.html) so I guess you need to set up a custom template as described [here](https://dotnet.github.io/docfx/tutorial/howto_create_custom_template.html)? That page doesn't mention this syntax option either. Is there a simple working example showing how to do this?

**Comment by Nukepayload2** [comment]
@jamaa 

> you need to set up a custom template

Yes. I've modified the `default` template and the `modern` template and uploaded them here: https://github.com/Nukepayload2/docfx-v2-vb-templates

They have both C# declarations and VB declarations. But I haven't added scripts and the combo box to filter declarations by programming languages yet.

> Is there a simple working example showing how to do this?

I've added docfx configuration to my DeepSeek API wrapper library.

https://github.com/Nukepayload2/DeepseekSdkVb/tree/master/DocFxConfig

To run the example, you need to build my fork of docfx with `Release` configuration. Then clone the `DeepseekSdkVb` library (including submodules). Finally run `docfx docfx.json --serve` in the window of `DocCmd.cmd`.

Note 1: the folder structure looks like this:

- [docfx](https://github.com/Nukepayload2/docfx/tree/feature/vb-declaration-in-v2)
- [DeepseekSdkVb](https://github.com/Nukepayload2/DeepseekSdkVb)

Note 2: if the path of `docfx` is not the local build, you need to run `dotnet tool uninstall -g docfx` first.
